plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -473,6 +473,29 @@ public void testDeletingEntirePartitionedTable()
         assertThat(this.loadTable(tableName).newScan().planFiles()).hasSize(0);
     }
 
+    @Test
+    public void testReadingSnapshotReference()
+    {
+        String tableName = "test_reading_snapshot_reference" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['regionkey']) AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = this.loadTable(tableName);
+        long refSnapshotId = icebergTable.currentSnapshot().snapshotId();
+        icebergTable.manageSnapshots()
+                .createTag("test-tag", refSnapshotId)
+                .createBranch("test-branch", refSnapshotId)
+                .commit();
+
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM tpch.tiny.nation LIMIT 5", 5);
+        assertQuery("SELECT * FROM " + tableName + " FOR VERSION AS OF " + refSnapshotId,
+                "SELECT * FROM nation");
+        assertQuery("SELECT * FROM " + tableName + " FOR VERSION AS OF 'test-tag'",
+                "SELECT * FROM nation");
+        assertQuery("SELECT * FROM " + tableName + " FOR VERSION AS OF 'test-branch'",
```

Comment:
I was looking over the following test

https://github.com/apache/iceberg/blob/e56c1cf8c16f527bf91b74399602138e215f4186/core/src/test/java/org/apache/iceberg/TestFastAppend.java#L525-L532

The newly introduced syntax would work to reference the latest snapshot in a branch.
How would we want to reference a specific snapshot from within a branch?

Reply:
> How would we want to reference a specific snapshot from within a branch?

If you want to reference a specific snapshot by ID, then you don't really need to specify the branch, the snapshot ID already uniquely identifies it.

If you want to do time travel within a branch, then yes it is not covered by this PR. We can use some way like `FOR VERSION AS OF 'branch@timestamp'`, but that feels a bit too hacky and requires further discussion. I am working on introducing a syntax like `SELECT * FROM table_name BRANCH branch_name` so you can do `SELECT * FROM table_name BRANCH branch_name FOR TIMESTAMP AS OF xxxx`. We can see how people like that once the PR is published.